workflow-bot-app/src/test/java/com/symphony/bdk/workflow/GetAttachmentIntegrationTest.java

```diff
@@ -0,0 +1,141 @@
+package com.symphony.bdk.workflow;
+
+import static com.symphony.bdk.workflow.custom.assertion.WorkflowAssert.assertThat;
+import static org.assertj.core.api.Assertions.fail;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.timeout;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.symphony.bdk.gen.api.model.V4AttachmentInfo;
+import com.symphony.bdk.gen.api.model.V4Message;
+import com.symphony.bdk.gen.api.model.V4Stream;
+import com.symphony.bdk.workflow.swadl.SwadlParser;
+import com.symphony.bdk.workflow.swadl.v1.Workflow;
+
+import lombok.extern.slf4j.Slf4j;
+import org.junit.jupiter.api.Test;
+
+import java.io.File;
+import java.nio.file.Paths;
+import java.util.Collections;
+import java.util.List;
+
+
+@Slf4j
+class GetAttachmentIntegrationTest extends IntegrationTest {
+
+  private static final String OUTPUTS_ATTACHMENT_PATH_KEY = "%s.outputs.attachmentPath";
+
+  @Test
+  void getAttachment() throws Exception {
+    final Workflow workflow =
+        SwadlParser.fromYaml(getClass().getResourceAsStream("/attachments/get-attachment.swadl.yaml"));
+    final String streamId = "STREAM_ID";
+    final String msgId = "MSG_ID";
+    final String attachmentId = "ATTACHMENT_ID";
+    final String attachmentName = "/attachments/myAttachment.txt";
+    final String expectedFilePath = Paths.get("src", "test", "resources").toFile().getAbsolutePath() + attachmentName;
+    final byte[] encodedBytes = mockBase64ByteArray();
+    final V4Message actualMessage = new V4Message();
+    final V4Stream v4Stream = new V4Stream();
+    final List<V4AttachmentInfo> attachments =
+        Collections.singletonList(new V4AttachmentInfo().id(attachmentId).name(attachmentName));
+    v4Stream.setStreamId("STREAM_ID");
+    actualMessage.setMessageId(msgId);
+    actualMessage.setStream(v4Stream);
+    actualMessage.setAttachments(attachments);
+
+    when(messageService.getMessage(msgId)).thenReturn(actualMessage);
+    when(messageService.getAttachment(streamId, msgId, attachmentId)).thenReturn(encodedBytes);
+
+    engine.deploy(workflow);
+
+    engine.onEvent(messageReceived("/get-attachment"));
+
+    verify(messageService, timeout(5000).times(1)).getMessage(msgId);
+    verify(messageService, timeout(5000).times(1)).getAttachment(streamId, msgId, attachmentId);
+
+    assertThat(workflow).isExecuted()
+        .hasOutput(String.format(OUTPUTS_ATTACHMENT_PATH_KEY, "getattachment"), expectedFilePath);
+    assertFileExistsAndDeleteIt(expectedFilePath);
```

Comment:
we should avoid using the file system while running tests
what we could do instead if have a store(InputStream) method in the resourceprovider interface and in our tests store it in memory so it can easily be retrieved and checked

Reply:
Actually we don't event need to store the files in tests. The method store(InputStream) stores the file and returns the its path which is the output of get-attachment , in the test we only returns the path without storing the file (assuming that the library method is working and does not need test)